packages/cozy-client/src/CozyClient.js

```diff
@@ -1784,6 +1785,25 @@ instantiation of the client.`
   async saveAfterFetchSettings(slug, itemsOrSetter, setterKeys) {
     return saveAfterFetchSettings(this, slug, itemsOrSetter, setterKeys)
   }
+
+  /**
+   * Invalidate a query
+   *
+   * This will fetch the query even if there is a fetchPolicy and replace data with the new one
+   * Can be useful after a mutation that don't provide the new data or doesn't have realtime updates
+   *
+   * @param {object} query - Query with definition and options
+   * @param {QueryDefinition} query.definition - Query Definition
+   * @param {import("./types").QueryOptions} query.options - Query Options
+   * @returns {Promise<import("./types").QueryState>} Query state
+   */
+  async invalidateQuery({ definition, options }) {
+    try {
+      return await this.query(definition, { ...options, invalidate: true })
+    } catch (error) {
+      throw error
+    }
+  }
```

Comment:
Also, I'm wondering if this should be an option of the query.

If we take a look at RQ, it seems that they have an `invalideQueries` method. https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation .

Reply:
Unfortunately, I'm not going to be able to do without it. The problem is when you move files from Nextcloud to Cozy. I have to make a request on the source folder to make the files disappear, which are now in Cozy under other ids.

Reply:
Having an `invalideQuery(id)` can be useful. But I don't think we should add it to the QueryOtpion. 

It should `reset` the redux store with the initial data for a query and that's it aka `status:pending ,data:[]` 

From a React Query perspective, the `invalidate` action add a new `isInvalidated` to the `QueryState`, and this `isInvalidated` is used alongside the `stale` concept. (https://github.com/TanStack/query/blob/a300d484750edb4b64bdefcc86abaac305d73b13/packages/query-core/src/query.ts#L331-L335)

Reply:
From what I understand, what you're describing with the initial return from the store corresponds more to `resetQueries` in React Query ([doc](https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientresetqueries)).

It's true that adding an `isInvalidated` parameter could be interesting without using QueryOptions. Especially as it would only allow the query to be refetched when requested via an `useQuery`.

The stale concept would not be similar to our `fetchPolicy` concept 🤔 We declare when we consider that the data will no longer be fresh and we need to fetch it back from the server. Invalidation is used to declare that the data is no longer fresh now.

Reply:
> The stale concept would not be similar to our fetchPolicy concept

I don't have access anymore to the paper, but I think that I've written something about fetchPolicy and stale. Not sure this is exactly the same concept. 

Invalidate can also "just" set the `lastFetch` to `-Infinity`.